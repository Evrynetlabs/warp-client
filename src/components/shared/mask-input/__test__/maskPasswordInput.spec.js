import React from 'react'

import { shallow } from 'enzyme'
import MaskPassInput from 'Components/shared/mask-input/MaskPasswordInput'

describe('MaskPasswordInput', () => {
  describe('When open mask input', () => {
    test('icon should contain slash', () => {
      const component = shallow(<MaskPassInput></MaskPassInput>)

      expect(component.find('.fa-eye-slash').length).toBe(1)
      expect(component).toMatchSnapshot()
    })
  })

  describe('When close mask input', () => {
    test('icon should contain slash', () => {
      const component = shallow(<MaskPassInput></MaskPassInput>)
      component.find('.MaskPasswordInput__mask__btn').simulate('click')
      expect(component.find('.fa-eye').length).toBe(1)
      expect(component).toMatchSnapshot()
    })
  })
})
