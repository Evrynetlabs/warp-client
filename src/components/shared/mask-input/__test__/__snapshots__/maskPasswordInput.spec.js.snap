// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaskPasswordInput When close mask input icon should contain slash 1`] = `
<Container
  className="p-0"
  fluid={false}
>
  <ForwardRef
    className="m-0 p-0"
    noGutters={false}
  >
    <Col
      className="p-0 MaskPasswordInput__input"
    >
      <FormControl
        maskiconprops={Object {}}
        type="text"
      />
    </Col>
    <Col
      className="p-0 MaskPasswordInput__mask"
    >
      <Button
        active={false}
        className="MaskPasswordInput__mask__btn"
        disabled={false}
        onClick={[Function]}
        type="button"
        variant=""
      >
        <i
          className="fas fa-eye text-white"
        />
      </Button>
    </Col>
  </ForwardRef>
</Container>
`;

exports[`MaskPasswordInput When open mask input icon should contain slash 1`] = `
<Container
  className="p-0"
  fluid={false}
>
  <ForwardRef
    className="m-0 p-0"
    noGutters={false}
  >
    <Col
      className="p-0 MaskPasswordInput__input"
    >
      <FormControl
        maskiconprops={Object {}}
        type="password"
      />
    </Col>
    <Col
      className="p-0 MaskPasswordInput__mask"
    >
      <Button
        active={false}
        className="MaskPasswordInput__mask__btn"
        disabled={false}
        onClick={[Function]}
        type="button"
        variant=""
      >
        <i
          className="fas fa-eye-slash text-white"
        />
      </Button>
    </Col>
  </ForwardRef>
</Container>
`;
